peps/pep-0758.rst

```diff
@@ -0,0 +1,432 @@
+is not necessary. Since ``.rim`` files are essentially changes only to the
+upload API, package resolvers and package installers can continue to function
+with the APIs they've always supported.
+
+External hosting resiliency
+===========================
+
+One of the key concerns leading to PEP 438's revocation in PEP 470 was
+potential user confusion when an external index disappeared. From PEP 470:
+
+   This confusion comes down to end users of projects not realizing if a
+   project is hosted on PyPI or if it relies on an external service. This
+   often manifests itself when the external service is down but PyPI is
+   not. People will see that PyPI works, and other projects works, but this
+   one specific one does not. They oftentimes do not realize who they need to
+   contact in order to get this fixed or what their remediation steps are.
+
+While the problem of external wheel hosting service going down is not directly
+solved by this PEP, several safeguards are in place to greatly reduce the
+potential burden on PyPI administrators.
+
+This PEP thus proposes that:
+
+- External wheel hosting is only allowed for packages which are owned by
+  `organization accounts <https://docs.pypi.org/organization-accounts/>`__.
+- Organization accounts do not automatically gain the ability to externally
+  host wheels; this feature MUST be explicitly enabled by PyPI admins. Since
+  this will not be a common request, we don't expect the overhead to be nearly
+  as burdensome as :pep:`541` resolutions, account recovery requests, or even
+  file/project size increase requests.  External hosting requests would be
+  handled in the same manner as those requests, i.e. via the `PyPI GitHub
+  support tracker <https://github.com/pypi/support>`__.
+- Organization accounts requesting external wheel hosting MUST register their
+  own support contact URI, be it a ``mailto`` URI for a contact email address,
+  or the URL to the organization's support tracker. Such a contact URI is
+  optional for organizations which do not avail themselves of external wheel
+  file hosting.
+
+Combined with the ``EXTERNAL-HOSTING`` file's ``Owner`` key, this allows for
+installer tools to unambiguously redirect any download errors away from the
+PyPI support admins and squarely to the organization's support admins.
+
+While the exact mechanics of storing and retrieving this organization support
+URL will be defined separately, for the sake of example, let's say a package
+``foo`` externally hosts wheel files on ```https://foo.example.com``
+<https://foo.example.com>`__ and that host becomes unreachable. When an
+installer tool tries to download and install the package ``foo`` wheel, the
+download step will fail. The installer would then be able to query PyPI to
+provide a useful error message to the end user:
+
+- The installer reads the ``Owner`` field from the ``EXTERNAL-HOSTING`` file
```

Comment:
```suggestion
- The installer downloads the `.rim` file and reads the ``Owner`` field from the ``EXTERNAL-HOSTING`` file
```

I think we should specifically call out that pypi.org should serve the RIM files.